gpii/node_modules/eventLog/src/eventLog.js

```diff
@@ -0,0 +1,396 @@
+/*
+ * Event logging.
+ *
+ * Copyright 2017 Raising the Floor - International
+ *
+ * Licensed under the New BSD license. You may not use this file except in
+ * compliance with this License.
+ *
+ * The R&D leading to these results received funding from the
+ * Department of Education - Grant H421A150005 (GPII-APCP). However,
+ * these results do not necessarily represent the policy of the
+ * Department of Education, and you should not assume endorsement by the
+ * Federal Government.
+ *
+ * You may obtain a copy of the License at
+ * https://github.com/GPII/universal/blob/master/LICENSE.txt
+ */
+
+"use strict";
+
+var fluid = require("infusion");
+var fs = require("fs"),
+    path = require("path");
+
+var gpii = fluid.registerNamespace("gpii");
+fluid.registerNamespace("gpii.eventLog");
+
+fluid.defaults("gpii.eventLog", {
+    gradeNames: ["fluid.component", "fluid.contextAware"],
+    contextAwareness: {
+        platform: {
+            checks: {
+                windows: {
+                    contextValue: "{gpii.contexts.windows}",
+                    gradeNames: "gpii.eventLog.windows"
+                }
+            }
+        }
+    },
+    components: {
+        installID: {
+            type: "gpii.installID"
+        }
+    },
+    invokers: {
+        logEvent: "gpii.eventLog.log",
+        logTimedEvent: "gpii.eventLog.logDuration"
+    },
+    listeners: {
+        "onCreate.installId": {
+            func: "gpii.eventLog.setInstallId",
+            args: ["@expand:{that}.installID.getInstallID()"]
+        },
+        "onCreate.log": {
+            func: "{that}.logEvent",
+            args: ["gpii", "Start", {}, "gpii-start"]
+        },
+        "onDestroy.log": {
+            func: "{that}.logTimedEvent",
+            args: ["gpii", "Stop", {}, "gpii-start"]
+        },
+        "{lifecycleManager}.events.onSessionStart": {
+            namespace: "eventLog",
+            func: "{that}.logEvent",
+            args: ["lifecycle", "SessionStart", {userToken: "{arguments}.1"}]
+        },
+        "{lifecycleManager}.events.onSessionStop": {
+            namespace: "eventLog",
+            func: "{that}.logTimedEvent",
+            args: [
+                "lifecycle",
+                "SessionStop",
+                {userToken: "{arguments}.1.model.userToken"},
+                "{arguments}.1.createTime",
+                "sessionTime"
+            ]
+        }
+    }
+});
+
+/**
+ * The timers for timed events.
+ */
+gpii.eventLog.timers = {};
```

Comment:
These should all be stored as members on the component rather than as globals. We do sometimes use globals of this kind but only when we are the victim of some other kind of architectural blunder.